### PR TITLE
fix(options): exclude `buildDir` for auto-imports

### PR DESCRIPTION
**Pull request by danielroe**
### 🔗 Linked issue

https://github.com/nuxt/framework/pull/5981, https://github.com/nuxt/bridge/issues/421

### ❓ Type of change

- [ ] 📖 Documentation (updates to the documentation or readme)
- [x] 🐞 Bug fix (a non-breaking change that fixes an issue)
- [ ] 👌 Enhancement (improving an existing functionality like performance)
- [ ] ✨ New feature (a non-breaking change that adds functionality)
- [ ] ⚠️ Breaking change (fix or feature that would cause existing functionality to change)

### 📚 Description

This disables processing the build output of the previous step from auto-imports. Credit to @cinob for spotting the issue; this just moves the fix up to Nitro if we think that makes sense.

cc: @antfu 

### 📝 Checklist

- [x] I have linked an issue or discussion.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

**Review by antfu — APPROVED** [review]

File: src/options.ts [hunk 1]
```diff
@@ -147,6 +147,13 @@ export async function loadOptions (userConfig: NitroConfig = {}): Promise<NitroO
   //     .map(i => new RegExp(`(^|\\/)${escapeRE(i.split('node_modules/').pop())}(\\/|$)(?!node_modules\\/)`))
   // ]
 
+  if (options.autoImport) {
+    options.autoImport.exclude = options.autoImport.exclude || [/[\\/]node_modules[\\/]/, /[\\/]\.git[\\/]/]
```
Comment (pi0) [inline src/options.ts:151]: Default config merging strategy is joining (not override) also we are already pushing buildDir. I think we can add node_modules and .git to the defaults as well.

Comment (danielroe) [reply src/options.ts:151]: Actually, are we sure it doesn't override?

https://github.com/unjs/unimport/blob/78e8d7416b2ea723167777cc3a1175109e850358/src/unplugin.ts#L26-L29

Comment (pi0) [reply src/options.ts:151]: Yes, I'm aware unimport does override. Nitro/Nuxt options extend in the other hands. Also good idea for duplicating defaults for predictable behavior 👍🏼

Comment (danielroe) [reply src/options.ts:151]: Oh, so you were suggesting we always push `.git` and `node_modules` as well?

Comment (danielroe) [reply src/options.ts:151]: Just checking 😉

**Review by pi0 — APPROVED** [review]

**Pull request merged**